src/SyncDb.ts

```diff
@@ -20,35 +24,93 @@ class SyncDb extends Command {
     'generate-connections': flags.boolean({ char: 'c', description: 'Generate connections' })
   };
 
+  /**
+   * Default CLI options for running synchronize.
+   *
+   * @param {*} userParams
+   * @returns {SyncParams}
+   */
+  getSyncParams(userParams: any): SyncParams {
+    return {
+      ...userParams,
+      // Individual success handler
+      onSuccess: (context: ExecutionContext) =>
+        printLine(`  [✓] ${context.connectionId} - Successful (${context.timeElapsed.toFixed(2)}s)`),
+
+      // Individual error handler
+      onFailed: (context: ExecutionContext) =>
+        printLine(`  [✖] ${context.connectionId} - Failed (${context.timeElapsed.toFixed(2)}s)`)
+    };
+  }
+
   /**
    * CLI command execution handler.
    *
    * @returns {Promise<void>}
    */
   async run(): Promise<void> {
+    // Set CLI environment as true.
+    process.env.SYNC_DB_CLI = 'true';
+
     const { flags: parsedFlags } = this.parse(SyncDb);
-    const params = {
-      force: parsedFlags.force
-    };
+    const params = this.getSyncParams({ ...parsedFlags });
 
     try {
-      await handleFlags(parsedFlags, {
-        info: this.log,
-        error: this.error
-      });
+      await handleFlags(parsedFlags);
+
       const config = await loadConfig();
       const connections = await resolveConnections();
 
-      const { synchronize } = await import('./migrator');
+      const { synchronize } = await import('./api');
+
+      await printLine('Synchronizing...\n');
 
-      await synchronize(config, connections, params);
-      process.exit(0);
+      const result = await synchronize(config, connections, params);
+
+      await this.displayResult(result);
     } catch (e) {
       log('Error caught: ', e, '\n');
-      this.error('An error occurred: ' + e);
+
+      await printError(e.toString());
+
       process.exit(-1);
     }
   }
+
+  /**
+   * Display the result.
+   *
+   * @param {SyncResult[]} result
+   */
+  async displayResult(result: SyncResult[]) {
+    const failed = result.filter(attempt => !attempt.success);
```

Comment:
Positive cases are easier to read.
 
```suggestion
    const success = results.filter(result => result.success);
```

Reply:
Yes, but we need all the failed attempts instead. For successful ones only the counts matter - for error cases we need full error trace. 

```ts
    failed.forEach(async (attempt, index) => {
      await printLine(`${index + 1}) ${attempt.connectionId}`);
      await printError(attempt.error);
      await printLine();
    });
```